app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  has_secure_password
+  validates :password, presence: true, length: {minimum: 6}
```

Comment:
1. Các giá trị như 6 255 50 thì khai báo vào file settings hì
để làm được thì thanh niên tìm hiểu `gem config in rails ` đi đã nhé

2. phân biệt giúp `downcase ` và `downcase!` có gì khác nhau hì